lldb/include/lldb/Core/ValueObject.h

```diff
@@ -668,6 +699,32 @@ class ValueObject {
   CreateValueObjectFromData(llvm::StringRef name, const DataExtractor &data,
                             const ExecutionContext &exe_ctx, CompilerType type);
 
+  static lldb::ValueObjectSP
+  CreateValueObjectFromBytes(lldb::TargetSP target_sp, const void *bytes,
+                             CompilerType type);
+
+  static lldb::ValueObjectSP CreateValueObjectFromBytes(lldb::TargetSP target,
+                                                        const void *bytes,
+                                                        lldb::BasicType type);
+
+  static lldb::ValueObjectSP CreateValueObjectFromAPInt(lldb::TargetSP target,
+                                                        const llvm::APInt &v,
+                                                        CompilerType type);
+
+  static lldb::ValueObjectSP
+  CreateValueObjectFromAPFloat(lldb::TargetSP target, const llvm::APFloat &v,
+                               CompilerType type);
+
+  static lldb::ValueObjectSP CreateValueObjectFromPointer(lldb::TargetSP target,
```

Comment:
How is this different from CreateValueObjectFromAddress?

Reply:
I agree, we already have this API unless this is doing things differently. That other API allows a name to be specified and is used for synthetic children, so you should be able to use it and specify no name

Reply:
"CreateValueObjectFromPointer" is different from "CreateValueObjectFromAddresss" In that the former creates a value object whose value is the pointer address; the latter creates a value object whose value is whatever the pointer points to (it dereferences the pointer).  I *can* update CreateValueObjectFromAddress to take a parameter telling it whether to dereference the pointer or use the pointer address as the value, if you would prefer.